main_test.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+var urlFlag = flag.String("url", "", "URL to run integration tests against")
+
+type APITestSuite struct {
```

Comment:
Why is this "API" when there is no API? Perhaps AppTestSuite? Or IntegrationTestSuite (longer and uglier but more precise)

Reply:
yeah, sorry, done.